File: jlpt/native_android/app/src/main/java/com/jlpt/android/domain/model/Exam.kt
package com.jlpt.android.domain.model

import java.time.Instant

data class ExamSummary(
    val id: Long,
    val level: JlptLevel,
    val title: String,
    val examType: ExamType,
    val skillType: SkillType?,
    val timeLimitMinutes: Int,
    val questionCount: Int
)

data class ChoiceOption(
    val id: Long,
    val choiceText: String,
    val displayOrder: Int
)

data class ExamQuestion(
    val id: Long,
    val skillType: SkillType,
    val questionText: String,
    val passageContent: String?,
    val listeningAudioId: Long?,
    val choices: List<ChoiceOption>
)

data class ExamDetail(
    val id: Long,
    val level: JlptLevel,
    val title: String,
    val examType: ExamType,
    val skillType: SkillType?,
    val timeLimitMinutes: Int,
    val questions: List<ExamQuestion>
)

data class Attempt(
    val id: Long,
    val examId: Long,
    val examTitle: String,
    val status: AttemptStatus,
    val startedAt: Instant,
    val submittedAt: Instant?,
    val score: Int?,
    val maxScore: Int?,
    val questions: List<ExamQuestion>
)

data class AnswerResult(
    val questionId: Long,
    val questionText: String,
    val selectedChoiceId: Long?,
    val correctChoiceId: Long,
    val correct: Boolean,
    val explanation: String?
)

data class AttemptResult(
    val attemptId: Long,
    val examId: Long,
    val score: Int?,
    val maxScore: Int?,
    val submittedAt: Instant?,
    val answers: List<AnswerResult>
)

File: jlpt/native_ios/Sources/Domain/Models/Exam.swift
import Foundation

struct ExamSummary: Identifiable, Equatable {
    let id: Int64
    let level: JlptLevel
    let title: String
    let examType: ExamType
    let skillType: SkillType?
    let timeLimitMinutes: Int
    let questionCount: Int
}

struct ChoiceOption: Identifiable, Equatable {
    let id: Int64
    let choiceText: String
    let displayOrder: Int
}

struct ExamQuestion: Identifiable, Equatable {
    let id: Int64
    let skillType: SkillType
    let questionText: String
    let passageContent: String?
    let listeningAudioId: Int64?
    let choices: [ChoiceOption]
}

struct ExamDetail: Identifiable, Equatable {
    let id: Int64
    let level: JlptLevel
    let title: String
    let examType: ExamType
    let skillType: SkillType?
    let timeLimitMinutes: Int
    let questions: [ExamQuestion]
}

struct Attempt: Identifiable, Equatable {
    let id: Int64
    let examId: Int64
    let examTitle: String
    let status: AttemptStatus
    let startedAt: Date
    let submittedAt: Date?
    let score: Int?
    let maxScore: Int?
    let questions: [ExamQuestion]
}

struct AnswerResult: Identifiable, Equatable {
    let questionId: Int64
    let questionText: String
    let selectedChoiceId: Int64?
    let correctChoiceId: Int64
    let correct: Bool
    let explanation: String?
    var id: Int64 { questionId }
}

struct AttemptResult: Identifiable, Equatable {
    let attemptId: Int64
    let examId: Int64
    let score: Int?
    let maxScore: Int?
    let submittedAt: Date?
    let answers: [AnswerResult]
    var id: Int64 { attemptId }
}
